website/static/js/fileBrowser.js

```diff
@@ -0,0 +1,1175 @@
+                    'id':  nodeId,
+                    'attributes': {
+                        'title': title
+                    }
+                }
+            };
+            var promise = m.request({method : 'PATCH', url : url, config : xhrconfig, data : data});
+            promise.then(function(result){
+                self.collectionMenuObject().item.label = title;
+                m.redraw(true);
+            });
+            self.dismissModal();
+            self.validation(false);
+            return promise;
+        };
+        self.applyDroppable = function _applyDroppable ( ){
+            $('.fb-collections ul>li').droppable({
+                hoverClass: 'bg-color-hover',
+                drop: function( event, ui ) {
+                    var collection = self.collections()[$(this).attr('data-index')];
+                    var dataArray = [];
+                    // If multiple items are dragged they have to be selected to make it work
+                    if (args.selected().length > 1) {
+                        args.selected().map(function(item){
+                            dataArray.push(buildCollectionNodeData(item.data.id));
+                        });
+                    } else {
+                        // if single items are passed use the event information
+                        dataArray.push(buildCollectionNodeData(ui.draggable.find('.title-text>a').attr('data-nodeID'))); // data-nodeID attribute needs to be set in project organizer building title column
+                    }
+                    function saveNodetoCollection (index) {
+                        function doNext (){
+                            if(dataArray[index+1]){
+                                saveNodetoCollection(index+1);
+                            }
+                            collection.data.count(collection.data.count()+1);
+                        }
+                        m.request({
+                            method : 'POST',
+                            url : collection.data.node.links.self + 'node_links/', //collection.data.node.relationships.linked_nodes.links.related.href,
+                            config : xhrconfig,
+                            data : dataArray[index]
+                        }).then(doNext, doNext); // In case of success or error. It doesn't look like mithril has a general .done method
+                    }
+                    if(dataArray.length > 0){
+                        saveNodetoCollection(0);
+                    }
+                }
+            });
+        };
+        self.validateName = function _validateName (val){
+            if (val === 'Bookmarks') {
+                self.validation(false);
+                self.validationError('Your collection name can\'t be "Bookmarks", because bookmarks feature requires this name. Please select any other name. ');
+            } else {
+                self.validationError('');
+                if(val.length > 0) {
```

Comment:
In general, avoid code that looks like `if(someBool) then assign otherBool to true else assign otherBool to false`. Simplify to

```js
self.validation(val.length > 0);
```